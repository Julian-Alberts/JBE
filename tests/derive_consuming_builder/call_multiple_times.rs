
use jbe::ConsumingBuilder;

#[derive(ConsumingBuilder, PartialEq, Debug)]
pub struct User {
    id: usize,
    name: String,
    email: Option<String>
}

fn main() {
    let builder = UserBuilder::default().email(String::from("jon@example.com"));
    let user = builder.build(10, String::from("Jon"));
    assert_eq!(user, User {
        id: 10,
        name: String::from("Jon"),
        email: Some(String::from("jon@example.com"))
    });
    let user = builder.build(11, String::from("Jon2"));
    assert_eq!(user, User {
        id: 11,
        name: String::from("Jon2"),
        email: Some(String::from("jon@example.com"))
    });
}