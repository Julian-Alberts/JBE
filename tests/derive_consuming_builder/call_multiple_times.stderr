error[E0382]: use of moved value: `builder`
  --> tests/derive_consuming_builder/call_multiple_times.rs:19:16
   |
12 |     let builder = UserBuilder::default().email(String::from("jon@example.com"));
   |         ------- move occurs because `builder` has type `UserBuilder`, which does not implement the `Copy` trait
13 |     let user = builder.build(10, String::from("Jon"));
   |                ------- value moved here
...
19 |     let user = builder.build(11, String::from("Jon2"));
   |                ^^^^^^^ value used here after move
